packages/eslint-plugin/src/rules/sort-type-constituents.ts

```diff
@@ -183,7 +191,7 @@ export default util.createRule<Options, MessageIds>({
     function checkSorting(
       node: TSESTree.TSIntersectionType | TSESTree.TSUnionType,
     ): void {
-      const sourceOrder = node.types.map(type => {
+      const sourceOrder = node.types.map((type): Constituent => {
```

Comment:
Nit: any particular reason to add the return type explicitly, here and in the other file?

Reply:
I wanted to make it clear what the `Constituent` was and where it came from. I ended up getting rid of `Constituent` when I extracted `requiresParentheses` into a utility function.